backend/src/agent/persistence/worker/persistence_worker.go

```diff
@@ -102,21 +102,16 @@ func (p *PersistenceWorker) RunWorker() {
 	}
 }
 
-// enqueue takes a ScheduledWorkflow and converts it into a namespace/name
-// string which is then put onto the work queue. This method should *not* be
-// passed resources of any type other than ScheduledWorkflow.
+// enqueue takes a Workflow or a ScheduledWorkflow (depends on the saver) and converts it
+// into a namespace/name string which is then put onto the work queue.
 func (p *PersistenceWorker) enqueue(obj interface{}) {
 	var key string
 	var err error
 	if key, err = cache.MetaNamespaceKeyFunc(obj); err != nil {
 		runtime.HandleError(fmt.Errorf("Equeuing object: error: %v: %+v", err, obj))
 		return
 	}
-	if p.enforceRequeueDelays {
-		p.workqueue.AddRateLimited(key) // Exponential backoff.
-	} else {
-		p.workqueue.Add(key) // For testing.
-	}
+	p.workqueue.Add(key)
```

Comment:
It was intended behavior before to use `AddRateLimited` when `enforceRequeueDelays==true`.
Can you explain why you always prefer immediately requeue? Do we have scalability concerns here?

Does it make sense to add an option to make `enforceRequeueDelays` configurable? Because if you change this line, the option is no longer used in this code base.

Reply:
OK, I found more details in https://github.com/kubeflow/pipelines/issues/4886. Let me take a look

Reply:
I see, there was another instance (on failure) where we use `enforceRequeueDelays`.
So this seems to make sense to me.

Workflow syncing will happen much more frequently after this change, because any argo workflow update will trigger an immediate requeue (if it's not already in queue).

However, I still have some concerns, because I believe the rate limiting was still intended. It's just that when everything is normal, we can use constant rate limiting, not exponential. Is that an option?

Another possible improvement is prioritizing type of updates, if a workflow changed from `running` state to one of the final states (or activeDeadlineSeconds is changed to 0), then this update should be of highest priority. Other updates during workflow run can be rate limited, because they might be outdated soon.

Reply:
Rate Limit can also be used to avoid hot loop. I'm not sure why at first we use `enforceRequeueDelays` for persistent agent. Will it cause performance issues when there are too many work items in the work queue? Any backgound info @Bobgy 
 If we think the delay is too much, in our code, the base delay is 10 seconds. Based on the book `programming kubernetes`, https://www.google.com/books/edition/Programming_Kubernetes/6VKjDwAAQBAJ?hl=en&gbpv=1&bsq=work%20queue. `Most controllers will just use the DefaultControllerRateLimiter() whose exponential back-ff starting at 5ms and going up to 1000 seconds.

If we think the 10 seconds delay is too much, I think we follow the behavior of DefaultControllerRateLimiter() ti change the exponential back-ff starting at 5ms.

Reply:
Continuing the discussion here: https://github.com/kubeflow/pipelines/issues/4886#issuecomment-754766153